pkg/protocols/grpcprot/trailer.go

```diff
@@ -0,0 +1,92 @@
+/*
```

Comment:
I do a reactor: type `trailer` is an alias for `header`.
`trailer` and `header` are two independent concepts in [gRPC protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md), so to avoid confusion, I recommend using two separate types.

Reply:
`trailer` is a concept of `http`, you may check the standard Go library, it is using `header` as `trailer`, we should follow it:

https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/net/http/request.go;l=277